lib/collection/property-list.js

```diff
@@ -229,25 +235,29 @@ _.extend(PropertyList.prototype, /** @lends PropertyList.prototype */ {
      * Iterate on each item of this list
      */
     each: function (iterator, context) {
-        _.each(this.members, iterator, context || this.__parent);
+        _.forEach(this.members, iterator.bind(context || this));
     },
 
     /**
-     * @param {Function} rule
+     * @param {Function} iterator
      */
-    filter: function (rule, context) {
-        return _.filter(this.members, rule, context);
+    filter: function (iterator, context) {
+        return _.filter(this.members, iterator.bind(context || this));
```

Comment:
This change should ideally not cause any issue. What I would rather love is that code is modified to not do binding at all in case context is falsey. That should make things performant.